crates/bevy_ecs/src/archetype.rs

```diff
@@ -786,12 +786,7 @@ pub struct Archetypes {
 }
 
 /// Metadata about how a component is stored in an [`Archetype`].
-pub struct ArchetypeRecord {
-    /// Index of the component in the archetype's [`Table`](crate::storage::Table),
-    /// or None if the component is a sparse set component.
-    #[expect(dead_code, reason = "This field is written to, but never read.")]
-    pub(crate) column: Option<usize>,
-}
+pub struct ArchetypeRecord {}
```

Comment:
Per Periwink on Discord (<https://discord.com/channels/691052431525675048/749335865876021248/1328528070323208273>):

> It was supposed to be used for the move toward FragmentingRelation and the work to add a ComponentIndex. If it's un-read right now it can probably be removed for now and re-added later